packages/react-composites/src/composites/CallComposite/pages/ConfigurationPage.tsx

```diff
@@ -303,3 +268,181 @@ const localPreviewTrampoline = (mobileView: boolean, doNotShow?: boolean): JSX.E
   }
   return <LocalPreview mobileView={mobileView} showDevicesButton={mobileView} />;
 };
+
+const cameraPermissionGrantedTrampoline = (
+  cameraPermissionGranted: boolean | undefined,
+  videoState?: PermissionState | undefined
+): boolean | undefined => {
+  /* @conditional-compile-remove(call-readiness) */
+  return videoState ? videoState === 'granted' : cameraPermissionGranted;
+
+  return cameraPermissionGranted;
+};
+
+const micPermissionGrantedTrampoline = (
+  microphonePermissionGranted: boolean | undefined,
+  audioState?: PermissionState | undefined
+): boolean | undefined => {
+  /* @conditional-compile-remove(call-readiness) */
+  return audioState ? audioState === 'granted' : microphonePermissionGranted;
+
+  return microphonePermissionGranted;
+};
+
+/* @conditional-compile-remove(call-readiness) */
+const callReadinessModal = (
```

Comment:
can we move this to its own file? ConfigurationPage is getting long

Reply:
made changes and pushed to PR https://github.com/Azure/communication-ui-library/pull/2529